### PR TITLE
Add C++11 keywords

### PR DESCRIPTION
**Pull request by jwillemsen**
* lib/coderay/scanners/cpp.rb:

### CODE REVIEW

**Comment by jwillemsen** [comment]
The travis failure looks independent of my changes, rake doesn't support the old ruby versions

**Comment by jwillemsen** [comment]
Yes, I can add some example code but I can't find the test for cpp, could you maybe point me to it?

**Comment by korny** [comment]
They're here: https://github.com/rubychan/coderay-scanner-tests/tree/master/cpp. Instructions are here: https://github.com/rubychan/coderay/wiki/Test-Suite

**Comment by jwillemsen** [comment]
Done, will create a separate pull request

**Comment by jwillemsen** [comment]
Added rubychan/coderay-scanner-tests#8

**Comment by jwillemsen** [comment]
Looks the failing check has nothing todo with my change

**Comment by korny** [comment]
Failing tests should be fixed on master, can you update your branch again please?

**Comment by jwillemsen** [comment]
Done, thanks for resolving this on master, now this is again green!

**Comment by jwillemsen** [comment]
Anything you need in addition from my side?

**Comment by jwillemsen** [comment]
Fixed merge conflict

**Comment by jwillemsen** [comment]
Ok, travis-ci now allows again C extensions but looks RedCloth has another problem :-(

**Comment by jwillemsen** [comment]
Ok, allowed failure for jruby18/19 because of the RedCloth problem, now this pull request is passed on Jenkins and has no conflicts

File: lib/coderay/scanners/cpp.rb [hunk 1]
```diff
@@ -2,43 +2,45 @@ module CodeRay
...
     title 'C++'
-    
+
     #-- http://www.cppreference.com/wiki/keywords/start
     KEYWORDS = [
       'and', 'and_eq', 'asm', 'bitand', 'bitor', 'break',
       'case', 'catch', 'class', 'compl', 'const_cast',
       'continue', 'default', 'delete', 'do', 'dynamic_cast', 'else',
-      'enum', 'export', 'for', 'goto', 'if', 'namespace', 'new',
+      'enum', 'export', 'final', 'for', 'goto', 'if', 'namespace', 'new',
```
Comment (korny) [inline lib/coderay/scanners/cpp.rb]: `final` is a directive, I think.

**Comment by korny** [comment]
Sorry for the delay. I also updated the list of Rubies in the Travis config file, so master is green now. Can you merge again, please?

**Comment by jwillemsen** [comment]
Merged upstream master again into my tree, resolved merge conflicts

**Comment by korny** [comment]
Thank you :)

**Comment by jwillemsen** [comment]
Thanks for accepting them!

**Pull request merged**